datasets/ted_talks_iwslt_2/ted_talks_iwslt_2.py

```diff
@@ -0,0 +1,417 @@
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(path_to_manual_file, f"xml/ted_{language}-20160408.zip"),
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(path_to_manual_file, f"xml-20150616/ted_{language}-20150530.zip"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+
+        source = filepath
+        language = self.config.language
+        print(filepath)
+
+        def et_to_dict(tree):
+            """This is used to convert the xml to a list of dicts"""
+
+            dct = {tree.tag: {} if tree.attrib else None}
+            children = list(tree)
+            if children:
+                dd = defaultdict(list)
+                for dc in map(et_to_dict, children):
+                    for k, v in dc.items():
+                        dd[k].append(v)
+                dct = {tree.tag: dd}
+            if tree.attrib:
+                dct[tree.tag].update((k, v) for k, v in tree.attrib.items())
+            if tree.text:
+                text = tree.text.strip()
+                if children or tree.attrib:
+                    if text:
+                        dct[tree.tag]["text"] = text
+                else:
+                    dct[tree.tag] = text
+            return dct
+
+        def get_talks(zipfile):
+            filename = zipfile.namelist()[0]
+            tree_source = ET.parse(zf_source.open(filename))
+            root_source = tree_source.getroot()
+            source_talks = et_to_dict(root_source).get("xml").get("file")
+
+            return source_talks
+
+        if os.path.exists(source):
+            with zipfile.ZipFile(source) as zf_source:
+                try:
+                    source_talks = get_talks(zf_source)
+                    # source_ids = [talk.get("head")[0].get("talkid")[0] for talk in source_talks]
+
+                except Exception as pe:
+                    print(f"ERROR: {pe}")
+                    print(f"Which likely means that you have a malformed XML file:: {source}\n")
```

Comment:
You can use `logging.warning` instead of `print`

Reply:
ha ha ...okay!